### PR TITLE
feat(DateInput): add `disablePortal` for use inside popover

### LINKED ISSUE #1010: [Popover] DateInput in popover causes it to close after clicking on calendar selector
### Component/Module
Popover
### Screenshots (if applicable)

https://github.com/narmi/design_system/assets/71462029/2975aa63-9bf3-4660-8af4-a8bef5f97ffa

### Reproduction Steps
1. create popover with content include DateInput with calendar selector for date
2. select date from calendar selector
### Expected Result
popover stays open
### Actual Result
popover closes
### Additional Details / Context
probably want some way to override if the popover is open. The issue would be how to still trigger if the user clicks off of the popover.

### PR DESCRIPTION
**Pull request by akdetrick**
fixes #1010 

Adds `disablePortal` prop to `DateInput` to append the calendar _inside_ its parent instead of `document.body`. This is useful when we want to use a `DateInput` inside a `Popover`.

Screen Shot 2023-08-29 at 6 51 50 PM
Screen Shot 2023-08-29 at 6 51 59 PM

### CODE REVIEW

File: src/DateInput/index.js [hunk 1]
```diff
@@ -56,21 +57,26 @@ const DateInput = ({
   };
 
   useEffect(() => {
+    if (disablePortal) {
+      console.dir({ ...props });
```
Comment (akdetrick) [inline src/DateInput/index.js:61]:
```suggestion
```

File: src/DateInput/index.js [hunk 2]
```diff
@@ -56,21 +57,25 @@ const DateInput = ({
   };
 
   useEffect(() => {
+    if (disablePortal) {
+      flatpickrOptions.static = true;
```
Comment (matthew-hogains) [inline src/DateInput/index.js:61]: Console statement

File: src/DateInput/index.js [hunk 3]
```diff
@@ -56,21 +57,25 @@ const DateInput = ({
...
-        ref={input}
-        type="date"
-        required
-        placeholder={props.label}
-        data-testid={testId}
-        {...props}
-      />
-    </Input>
+    <div>
+      <Input {...props} label={props.label}>
```
Comment (akdetrick) [inline src/DateInput/index.js:68]: static mode of flatpickr does some weird things with the input placeholder - moving the label prop to the `Input` component makes things behave more normal regardless of the flatpickr mode

File: src/DateInput/index.js (hunk 2 above)
Comment (akdetrick) [reply src/DateInput/index.js:61]: fixed

**Pull request merged**